chain/network/src/network_protocol/testonly.rs

```diff
@@ -299,10 +298,16 @@ impl Chain {
         self.blocks.iter().map(|b| b.header().clone()).collect()
     }
 
-    pub fn make_config(&self, port: u16) -> NetworkConfig {
+    pub fn make_config<R: Rng>(&self, rng: &mut R) -> config::NetworkConfig {
         // TODO(gprusak): make config generation rng-based,
         // rather than using a seed.
-        NetworkConfig::from_seed("test1", port)
+        let port = crate::test_utils::open_port();
```

Comment:
I'd maybe still pass the `port` in -- in the ideal state, the caller would open a socket first, to avoid races.

Reply:
This is testonly and it changes all the time, as I write/migrate more tests and align the utils accordingly, so it doesn't really matter. My current plan is to eventually add a TcpSocket to the NetworkConfig. Since TcpSocket will be then consumed by the constructor, it will probably require splitting the config into immutable and mutable parts.